### PR TITLE
Galleon feature  pack 

### PR DESCRIPTION
**Pull request by jimma**

### CODE REVIEW

File: modules/dist/pom.xml [hunk 1]
```diff
@@ -93,44 +46,50 @@
...
-                      <type>zip</type>
-                      <overWrite>false</overWrite>
-                      <outputDirectory>${project.build.directory}</outputDirectory>
-                    </artifactItem>
-                  </artifactItems>
-                </configuration>
-              </execution>
-            </executions>
-          </plugin>
+          <plugins>
```
Comment (jbliznak) [inline modules/dist/pom.xml:57]: this has different indentation than previously

File: modules/feature-pack/pom.xml [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.jboss.shrinkwrap</groupId>
+            <artifactId>shrinkwrap-depchain</artifactId>
+            <type>pom</type>
+            <scope>provided</scope>
+        </dependency>
```
Comment (jbliznak) [inline modules/feature-pack/pom.xml:80]: do we need those two dependencies in this module? It doesn't seem it triggers any tests

File: modules/feature-pack/src/main/resources/modules/system/layers/base/org/jboss/ws/cxf/jbossws-cxf-client/main/module.xml [hunk 3]
```diff
@@ -20,31 +21,32 @@
...
         <module name="javax.xml.bind.api" export="true"/>
-        <module name="com.sun.xml.bind" export="true" services="export"/>
+        <module name="org.glassfish.jaxb" export="true" services="export"/>
         <module name="javax.xml.ws.api" export="true"/>
         <module name="javax.jws.api" export="true"/>
         <!-- WSDL4J API -->
         <module name="javax.wsdl4j.api" export="true"/>
         <!-- JBossWS API -->
-        <module name="org.jboss.ws.api" export="true"/>
+        <module name="org.jboss.ws.api" export="true" />
```
Comment (jbliznak) [inline modules/feature-pack/src/main/resources/modules/system/layers/base/org/jboss/ws/cxf/jbossws-cxf-client/main/module.xml:36]: this is just a formatting issue, but some changed lines add this space before closing tag while the rest is kept as is. It isn't a functional issue but it doesn't look good to have the mix of two formats

Comment (jbliznak) [reply modules/feature-pack/src/main/resources/modules/system/layers/base/org/jboss/ws/cxf/jbossws-cxf-client/main/module.xml:36]: this applies for all module XML files in this PR.

File: modules/dist/pom.xml (hunk 1 above)
Comment (jimma) [reply modules/dist/pom.xml:57]: Fixed.

File: modules/feature-pack/pom.xml (hunk 2 above)
Comment (jimma) [reply modules/feature-pack/pom.xml:80]: You are right. This should be removed.

File: modules/feature-pack/src/main/resources/modules/system/layers/base/org/jboss/ws/cxf/jbossws-cxf-client/main/module.xml (hunk 3 above)
Comment (jimma) [reply modules/feature-pack/src/main/resources/modules/system/layers/base/org/jboss/ws/cxf/jbossws-cxf-client/main/module.xml:36]: I already formatted all items.

File: pom.xml [hunk 4]
```diff
@@ -1035,6 +1039,16 @@
...
       </plugin>
+      <plugin>
+        <groupId>org.jboss.galleon</groupId>
+        <artifactId>galleon-maven-plugin</artifactId>
+        <version>${org.jboss.galleon.version}</version>
+      </plugin>
+       <plugin>
+         <groupId>org.wildfly.galleon-plugins</groupId>
+         <artifactId>wildfly-galleon-maven-plugin</artifactId>
+         <version>6.3.1.Final</version>
```
[the remark this answers is not in the document]
Comment (jimma) [reply pom.xml:1050]: Definitely. Fixed.

**Comment by jimma** [comment]
@jbliznak  Thanks a lot for the review. I already fixed all the places you spotted. Please review it again.

File: modules/dist/pom.xml [hunk 5]
```diff
@@ -158,62 +117,6 @@
...
-                      <phase>prepare-package</phase>
-                      <goals>
-                          <goal>single</goal>
-                      </goals>
-                      <configuration>
-                          <finalName>assembly</finalName>
-                          <ignoreDirFormatExtensions>true</ignoreDirFormatExtensions>
-                          <appendAssemblyId>false</appendAssemblyId>
-                          <descriptors>
-                              <descriptor>src/main/scripts/assembly-deploy-artifacts.xml</descriptor>
```
Comment (jbliznak) [inline modules/dist/pom.xml:189]: so this PR should contain removal of this file too?

File: modules/dist/pom.xml [hunk 6]
```diff
@@ -158,62 +117,6 @@
...
     </profile>
-    <!--
-    Name:  deploy
-    Descr: Deploy stack on WildFly instance
-    -->
-    <profile>
-      <id>deploy</id>
-      <activation>
-        <property>
-          <name>!nodeploy</name>
```
Comment (jbliznak) [inline modules/dist/pom.xml:169]: README should be updated on this parameter

File: modules/dist/pom.xml [hunk 7]
```diff
@@ -158,62 +117,6 @@
...
-              <execution>
-                <id>deploy</id>
-                <phase>prepare-package</phase>
-                <goals>
-                  <goal>run</goal>
-                </goals>
-                <configuration>
-                  <target>
-                    <property name="jboss.home" value="${jboss.home}" />
-                    <ant antfile="src/main/scripts/build-deploy.xml" />
```
Comment (jbliznak) [inline modules/dist/pom.xml:208]: another file to remove?

File: modules/feature-pack/src/main/resources/modules/system/layers/base/org/jboss/ws/cxf/jbossws-cxf-client/main/module.xml (hunk 3 above)
Comment (jbliznak) [reply modules/feature-pack/src/main/resources/modules/system/layers/base/org/jboss/ws/cxf/jbossws-cxf-client/main/module.xml]: There are still some inconsistencies, see:
```
grep -r " />" --include=module.xml
```

- https://github.com/jimma/jbossws-cxf/blob/galleon/modules/feature-pack/src/main/resources/modules/system/layers/base/org/jboss/ws/common/main/module.xml#L42
- https://github.com/jimma/jbossws-cxf/blob/galleon/modules/feature-pack/src/main/resources/modules/system/layers/base/org/jboss/ws/cxf/jbossws-cxf-server/main/module.xml
- https://github.com/jimma/jbossws-cxf/blob/galleon/modules/feature-pack/src/main/resources/modules/system/layers/base/org/jboss/ws/jaxws-client/main/module.xml

File: modules/dist/pom.xml (hunk 6 above)
Comment (jbliznak) [reply modules/dist/pom.xml:169]: also docs https://github.com/jimma/jbossws-cxf/blob/galleon/docbook/src/main/doc/adoc/content/chapter-8-Build_and_testsuite_framework.adoc?plain=1#L127

File: modules/dist/pom.xml (hunk 5 above)
Comment (jimma) [reply modules/dist/pom.xml:189]: Yes. This is replaced with the jbossws-galleon-feature pack.

File: modules/dist/pom.xml (hunk 7 above)
Comment (jimma) [reply modules/dist/pom.xml:208]: Right. This is replaced by the server provisioning with org.jboss.galleon:galleon-maven-plugin.

File: modules/dist/pom.xml (hunk 6 above)
Comment (jimma) [reply modules/dist/pom.xml:169]: Fixed. Thanks.

**Pull request merged**